### PR TITLE
feature(widgets): added isset on \ElggWidget objects to check settings

### LINKED ISSUE #9590: 2.x: ElggWidget needs __isset to match __unset
Needs to checks private settings.

### PR DESCRIPTION
**Pull request by jdalsem**
fixes #9590

### CODE REVIEW

File: engine/classes/ElggWidget.php [hunk 1]
```diff
@@ -113,6 +113,27 @@ public function __unset($name) {
...
+	 *
+	 * @return bool
+	 * @since 2.2.0
+	 */
+	public function __isset($name) {
+		if (array_key_exists($name, $this->attributes)) {
+			return parent::__isset($name);
+		} else {
+			$private_setting = $this->getPrivateSetting($name);
+			return is_null($private_setting);
```
Comment (mrclay) [inline engine/classes/ElggWidget.php]: `!is_null($private_setting)`

Comment (mrclay) [inline engine/classes/ElggWidget.php]: Is there a test suite for ElggWidget? We should add a basic test here.

Comment (jdalsem) [inline engine/classes/ElggWidget.php]: Added #9593 to add the tests (as we there are none for ElggWidget at the moment)
